forms-flow-idm/keycloak/docker-compose.yml

```diff
@@ -25,26 +25,26 @@ services:
       - keycloak-server-network
 
   keycloak:
-    image: quay.io/keycloak/keycloak:14.0.0
+    image: quay.io/keycloak/keycloak:21.1.1
     restart: unless-stopped
     container_name: keycloak
     volumes:
-      - ./imports:/opt/jboss/keycloak/imports
-      - ./themes/formsflow:/opt/jboss/keycloak/themes/formsflow
-      
+      - ./imports:/opt/keycloak/data/import
+      - ./themes/formsflow:/opt/keycloak/themes/formsflow
     command: 
-      - "-b 0.0.0.0 -Dkeycloak.import=/opt/jboss/keycloak/imports/formsflow-ai-realm.json -Dkeycloak.migration.strategy=OVERWRITE_EXISTING"
+      - "start-dev --import-realm --http-relative-path /auth"
```

Comment:
so when I deploy this container it will take the default roles and user details from import folder and will create automatically  right ?

Reply:
Yes, that's correct

Reply:
![image](https://github.com/AOT-Technologies/forms-flow-ai/assets/95394061/06d34949-1719-473a-8b63-9cf73c377c0a)
I tested this branch its working fine

Reply:
@shuhaib-aot @sumesh-aot  You should not use the `start-dev` in production. Instead use just `start` and remove `--optimized` option, since the dev server is like the name says only for development. The startup will be slower but keycloak detects the change automatically and rebuilds the new "quarkus image" with the needed `--http-relative-path` option enabled.

You may also use `--optimized` after the initial `build` but the configuration has to be persisted as described in the [keycloak docs](https://www.keycloak.org/server/configuration#_optimize_the_keycloak_startup)

Reply:
@PeeK1e to start keycloak in production mode; it needs a production setup such as keys/certificates (mandated by new versions), which would be specific to each client installation. That's the reason, I added start-dev to get the development going on, and change the compose file as needed for any implementation. 
Just wondering, were you able to run keycloak without any key/cert configuration in production mode ?

Reply:
Are you talking about SSL Certificates?
We're almost always using a SSL Terminated Ingress (even for testing purposes). So we don't need any certs if you are talking about them.

Reply:
@PeeK1e Intention of this PR is to make supporting changes on the product side to support Keycloak v21. The keycloak startup docker-compose; we can add only in a way were it works for everyone. Keycloak to run in a production mode; it enforces SSL, and not every developer setup would be having it. 
Anyone setting up Keycloak for production setup should follow keycloak documentation and make necessary changes to the docker-compose.

Reply:
@gbates101 the deployment templates in this repo are for dev purpose (local development), we will be updating the templates in https://github.com/AOT-Technologies/forms-flow-ai-deployment for non-dev deployments.